test/commands/git/sync_track_test.rb

```diff
@@ -0,0 +1,41 @@
+require "test_helper"
+
+class Git::SyncTrackTest < ActiveSupport::TestCase
+  test "no change when git SHA matches HEAD SHA" do
+    track = create :track, slug: 'fsharp', git_sha: "HEAD"
+
+    Git::SyncTrack.(track)
+
+    refute track.changed?
+  end
+
+  test "no change when there are no track changes in the diff" do
+    track = create :track, slug: 'fsharp', git_sha: "765f921ce85917cfe22b2a608d370f67da57820b"
+
+    Git::SyncTrack.(track)
+
+    refute track.changed?
+  end
+
+  test "git SHA changes to HEAD SHA when there are changes" do
+    track = create :track, slug: 'fsharp', active: false, git_sha: "f290a29144b93b21e2399cd532b22562d83b6a52"
+
+    Git::SyncTrack.(track)
+
+    assert_equal "70a8473aa074c1ee9c843665347481fe5c8ee95f", track.git_sha
```

Comment:
I wonder if it's worth on this line reading the `"70a..."` from git itself. Otherwise whenever the repo changes in the tests, this test is going to break.